src/sqlancer/common/oracle/CERTOracleBase.java

```diff
@@ -0,0 +1,88 @@
+package sqlancer.common.oracle;
+
+import java.util.Arrays;
+import java.util.List;
+
+import sqlancer.Randomly;
+import sqlancer.SQLGlobalState;
+import sqlancer.common.query.ExpectedErrors;
+
+public abstract class CERTOracleBase<S extends SQLGlobalState<?, ?>> implements TestOracle<S> {
+
+    protected final S state;
+    protected final ExpectedErrors errors;
+    protected List<String> queryPlan1Sequences;
+    protected List<String> queryPlan2Sequences;
+
+    protected enum Mutator {
+        JOIN, DISTINCT, WHERE, GROUPBY, HAVING, AND, OR, LIMIT;
+
+        public static Mutator getRandom(Mutator... exclude) {
```

Comment:
The method name is confusing. Could we name this `getRandomExcept`, `getRandomExcluding`, or something similar? Currently, I'd believe that we randomly select from one of the options.